File: frontend/solutions/p6/aiSolution.py
class Solution:
    def isArraySpecial(self, nums: List[int], queries: List[List[int]]) -> List[bool]:
        n = len(nums)
        prefix = [0] * n
        for i in range(1, n):
            prefix[i] = prefix[i - 1] + (0 if nums[i] % 2 != nums[i - 1] % 2 else 1)

        return [prefix[r] - prefix[l] == 0 for l, r in queries]

File: frontend/solutions/p6/aiSolution.js
/**
 * @param {number[]} nums
 * @param {number[][]} queries
 * @return {boolean[]}
 */
var isArraySpecial = function(nums, queries) {
    const n = nums.length;
    const prefix = new Array(n).fill(0);
    for (let i = 1; i < n; i++) {
        prefix[i] = prefix[i - 1] + (nums[i] % 2 !== nums[i - 1] % 2 ? 0 : 1);
    }

    return queries.map(([l, r]) => prefix[r] - prefix[l] === 0);
};